controllers/change_coordinators_test.go

```diff
@@ -40,79 +42,285 @@ var _ = Describe("Reconcile.SelectCandidates", func() {
+				Expect(cluster.DesiredCoordinatorCount()).To(BeNumerically("==", 3))
+				Expect(len(candidates)).To(BeNumerically("==", cluster.DesiredCoordinatorCount()))
+
+				// Only select Storage processes since we select 3 processes and we have 4 storage processes
+				for _, candidate := range candidates {
+					Expect(candidate.ID).NotTo(Equal("storage-2"))
+					Expect(strings.HasPrefix(candidate.ID, "storage")).To(BeTrue())
+				}
+			})
 		})
 
-		BeforeEach(func() {
-			// Put the bad data in place
-			// storage - are there symbols for these ips and ids?
-			adminClient.ExcludedAddresses = append(adminClient.ExcludedAddresses, "1.1.0.1")
-			// log
-			adminClient.ExcludedAddresses = append(adminClient.ExcludedAddresses, "1.1.5.1")
-			cluster.Status.PendingRemovals = map[string]fdbtypes.PendingRemovalState{
-				"operator-test-1-log-2":     {},
-				"operator-test-1-storage-2": {},
+		When("when multiple storage process are marked for removal", func() {
+			removals := []string{
+				"storage-2",
+				"storage-3",
 			}
-			Expect(cluster.InstanceIsBeingRemoved("operator-test-1-log-2")).To(BeTrue())
+
+			BeforeEach(func() {
+				cluster.Spec.InstancesToRemove = removals
+				for _, removal := range removals {
+					Expect(cluster.InstanceIsBeingRemoved(removal)).To(BeTrue())
+				}
+			})
+
+			It("should select 2 storage processes and 1 TLog", func() {
+				Expect(cluster.DesiredCoordinatorCount()).To(BeNumerically("==", 3))
+				Expect(len(candidates)).To(BeNumerically("==", cluster.DesiredCoordinatorCount()))
+
+				// Only select Storage processes since we select 3 processes and we have 4 storage processes
+				storageCnt := 0
+				logCnt := 0
+				for _, candidate := range candidates {
+					for _, removal := range removals {
+						Expect(candidate.ID).NotTo(Equal(removal))
+					}
+
+					if strings.HasPrefix(candidate.ID, "storage") {
+						storageCnt++
+					}
+
+					if strings.HasPrefix(candidate.ID, "log") {
+						logCnt++
+					}
+				}
+
+				Expect(storageCnt).To(BeNumerically("==", 2))
+				Expect(logCnt).To(BeNumerically("==", 1))
+			})
 		})
+	})
+
+	When("Using a HA cluster with three DCs and two regions", func() {
+		var status *fdbtypes.FoundationDBStatus
+		var candidates []localityInfo
+
+		JustBeforeEach(func() {
+			cluster.Spec.UsableRegions = 2
+			cluster.Spec.DataCenter = "primary"
+
+			var err error
+			status, err = adminClient.GetStatus()
+			Expect(err).NotTo(HaveOccurred())
+
+			// generate status for 2 dcs and 1 sate
+			status.Cluster.Processes = generateProcessInfo(2, 1)
 
-		Context("with candidates", func() {
-			var candidates []localityInfo
-			JustBeforeEach(func() {
-				candidates = make([]localityInfo, 0, len(status.Cluster.Processes))
+			candidates, err = selectCoordinators(cluster, status)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("all processes are healthy", func() {
+			FIt("should only select storage processes in primary and remote and Tlog in satellite", func() {
```

Comment:
```suggestion
			It("should only select storage processes in primary and remote and Tlog in satellite", func() {
```

Reply:
Yeah, I still have to add some more tests to ensure the selection works also in some other weird cases or return the correct error. I actually thought I marked thee PR as draft 🤔